cranelift/codegen/src/isa/riscv64/inst/regs.rs

```diff
@@ -0,0 +1,245 @@
+//! Riscv64 ISA definitions: registers.
+//!
+
+use crate::machinst::ArgsOrRets;
+use crate::settings;
+
+use crate::machinst::{Reg, Writable};
+
+use crate::machinst::RealReg;
+use alloc::vec;
+use alloc::vec::Vec;
+
+use regalloc2::VReg;
+use regalloc2::{MachineEnv, PReg, RegClass};
+
+// first argument of function call
+#[inline(always)]
+pub fn a0() -> Reg {
+    x_reg(10)
+}
+
+// second argument of function call
+#[inline(always)]
+pub fn a1() -> Reg {
+    x_reg(11)
+}
+
+// third argument of function call
+#[inline(always)]
+pub fn a2() -> Reg {
+    x_reg(12)
+}
+#[inline(always)]
+pub fn a7() -> Reg {
+    x_reg(17)
+}
+
+#[inline(always)]
+pub fn param_or_rets_xregs(args_or_rets: ArgsOrRets) -> Vec<Writable<Reg>> {
```

Comment:
This should be implemented without dynamically allocating memory and building a `Vec`. Can you return a static `&[Writable<Reg>]` ?

Reply:
ok,I will try.

Reply:
look I cannot because of 
~~~
error[E0015]: cannot call non-const fn `inst::regs::x_reg` in constants
~~~

Reply:
I think  that `x_reg` could be made a const-fn, every function it calls is const or could be made const.

Reply:
@a1phyr yes
![image](https://user-images.githubusercontent.com/96557710/179336523-3828ac41-1191-46b9-8ced-1af47c69fe2a.png)
but some function is not a const.

Reply:
I have already remove the function "param_or_rets_xregs".